worker/changestream/stream/stream.go

```diff
@@ -221,7 +231,43 @@ func (s *Stream) loop() error {
 			select {
 			case <-s.tomb.Dying():
 				return tomb.ErrDying
-			case <-term.done:
+
+			case <-s.clock.After(defaultWaitTermTimeout):
+				// This is a critical error, we should never get here if juju
+				// is humming along. This is a sign that something is wrong
+				// with the dependencies of the worker. We have no choice but
+				// to return an error and to bounce the world.
+				return errors.Errorf("term has not been completed in time")
+
+			case empty, ok := <-term.done:
+				if !ok {
+					// If the event mux has been killed, then the term has been
+					// aborted, so we just continue. This is likely the case
+					// when the worker is dying. We don't want to block the
+					// change stream, so we just continue.
+					s.logger.Infof("term has been aborted")
+					continue
```

Comment:
Presumably, a stopped multiplexer that is not coming back will mean the next term dispatch hits the 30 second bound and we bounce ourselves.